### PR TITLE
 infra: add a job to automatically mint certs for vanity URLs 

### PR DESCRIPTION
**Pull request by stevekuznetsov**
/assign @hongkailiu

### CODE REVIEW

File: ci-operator/jobs/infra-periodics.yaml [hunk 1]
```diff
@@ -84,6 +84,57 @@ periodics:
...
+        set -o pipefail
+        oc whoami
+        oc project
+        export CLOUDSDK_CONFIG=/tmp/gcloudconfig
+        mkdir -p "${CLOUDSDK_CONFIG}"
+        gcloud auth activate-service-account --key-file="${GOOGLE_APPLICATION_CREDENTIALS}"
+        /acme/acme.sh --version
+        /acme/acme.sh --issue --dns dns_gcloud -d 'prow.ci.openshift.com' --home /data --dnssleep 60 --fullchain-file /data/prow.fullchain.cer --key-file /data/prow.key --force
+        ls -al /data
+        oc -n ci create secret tls prow-cert --cert=/data/prow.fullchain.cer --key=/data/prow.key --dry-run -o yaml | oc -n ci apply -f -
```
Comment (hongkailiu) [inline ci-operator/jobs/infra-periodics.yaml:110]: Who is going to consume this secret?

File: ci-operator/jobs/infra-periodics.yaml [hunk 2]
```diff
@@ -84,6 +84,57 @@ periodics:
...
+        set -o errexit
+        set -o nounset
+        set -o pipefail
+        oc whoami
+        oc project
+        export CLOUDSDK_CONFIG=/tmp/gcloudconfig
+        mkdir -p "${CLOUDSDK_CONFIG}"
+        gcloud auth activate-service-account --key-file="${GOOGLE_APPLICATION_CREDENTIALS}"
+        /acme/acme.sh --version
+        /acme/acme.sh --issue --dns dns_gcloud -d 'prow.ci.openshift.com' --home /data --dnssleep 60 --fullchain-file /data/prow.fullchain.cer --key-file /data/prow.key --force
```
Comment (hongkailiu) [inline ci-operator/jobs/infra-periodics.yaml:108]: Is `prow.ci.openshift.com` managed by our GCP account?

File: core-services/ci-secret-bootstrap/_config.yaml [hunk 3]
```diff
@@ -402,6 +402,17 @@
     - cluster: app.ci
       namespace: ci
       name: gce-sa-credentials-gcs-publisher
+- from:
+    service-account.json:
+      bw_item: cert-issuer
+      attachment: credentials.json
+  to:
+    - cluster: build01
```
Comment (hongkailiu) [inline core-services/ci-secret-bootstrap/_config.yaml:410]: Why is it on `build01`?

**Comment by alvaroaleman** [comment]
/hold
Lets use cert-manager instead

File: ci-operator/jobs/infra-periodics.yaml [hunk 4]
```diff
@@ -84,6 +84,57 @@ periodics:
...
+    automountServiceAccountToken: true
+    volumes:
+    - emptyDir: {}
+      name: data
+    - name: cluster-secrets-gcp
+      secret:
+        items:
+        - key: service-account.json
+          path: service-account.json
+        secretName: gce-sa-credentials-gcs-publisher
```
Comment (hongkailiu) [inline ci-operator/jobs/infra-periodics.yaml:137]: Should it be `gce-sa-credentials-cert-issuer`?

**Comment by stevekuznetsov** [comment]
Going to do this using cert-manger

/close

**Pull request closed without merging**